src/txn/txn.c

```diff
@@ -714,8 +856,14 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
          * Resolve the prepared update to be committed update.
          */
         __txn_resolve_prepared_update(session, upd);
+        resolved = true;
     }
 
+    /* Fix the history store contents if exists. */
+    if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY) && resolved &&
+      (upd == NULL || F_ISSET(upd, WT_UPDATE_HS)))
```

Comment:
When does a `NULL` `upd` require fixing up the history store?

Reply:
When a page gets evicted with prepared updates, only the prepared updates are restored back into the page when it gets read back, so we need to handle the cases. But now we are writing prepared updates only in eviction, so there is no need to check for `upd`, it always needs to go and fix the history store. But if there doesn't exist any history store records, it skips. This has some performance penalty for prepared inserts.